### PR TITLE
Rename `logger/init.h` to `logger/logger.h`

### LINKED ISSUE #267: Rename `logger/init.h` to `Logger.h`
**Describe the bug**
Due to how it evolved, we currrently define the logger in `logger/init.h`, we should rename this to `logger/Logger.h`.

**To Reproduce**
NA

**Expected behavior**
NA

**Screenshots**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by kelvinkoon**
### Description
Rename `logger/init.h` to `logger/logger.h`.

### Testing Done
Files built successfully.

### Resolved Issues

Resolves #267 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/hl/stp/action/intercept_ball_action.cpp [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #include "software/ai/evaluation/robot.h"
 #include "software/ai/intent/move_intent.h"
 #include "software/geom/util.h"
-#include "software/logger/init.h"
+#include "software/logger/Logger.h"
```
Comment (MathewMacDougall) [inline src/software/ai/hl/stp/action/intercept_ball_action.cpp]: so uh, minor nit: Filename should be lowercase `logger.h` according to our style guide :sweat_smile:

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]

File: src/software/ai/hl/stp/action/intercept_ball_action.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/software/ai/hl/stp/action/intercept_ball_action.cpp]: Oh yea I just realized it was capitalized in the ticket. Whoops, sorry about that.

Comment (kelvinkoon) [reply src/software/ai/hl/stp/action/intercept_ball_action.cpp]: :scream: Fixed it.

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by akhilveeraghanta — APPROVED** [review]
LGTM :tada:

**Pull request merged**